### PR TITLE
Explain Nginx setup for Docker (and bare metal for www)

### PR DESCRIPTION
**Pull request by melroy89**
- Explain how to use Nginx if you are running the Docker setup
- Add missing section about the `www` subdomain, remove `www` from the server section, this caused a lot of issues for @Jerry-infosecexchange I think.. Sorry.

### CODE REVIEW

File: docs/docker_deployment_guide.md [hunk 1]
```diff
@@ -170,12 +170,87 @@ If you created the file `compose.override.yml` with your configs (`cp compose.pr
...
+        proxy_set_header HOST $host;
+        proxy_set_header X-Forwarded-Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_pass http://127.0.0.1:8008;
+    }
+
+    location /.well-known/mercure {
+        proxy_pass http://127.0.0.1:8080$request_uri;
```
Comment (nobodyatroot) [inline docs/docker_deployment_guide.md:240]: Only thing here is the port needs changed to match above, 8008 not 8080.

**Review by nobodyatroot — CHANGES_REQUESTED** [review]

File: docs/docker_deployment_guide.md (hunk 1 above)
Comment (melroy89) [reply docs/docker_deployment_guide.md]: Oopsy. you are right! Fixed now. But nothing else? Can we remove the HTTPS upgrade section maybe?

File: docs/docker_deployment_guide.md [hunk 2]
```diff
@@ -170,12 +170,87 @@ If you created the file `compose.override.yml` with your configs (`cp compose.pr
...
+    add_header Strict-Transport-Security "max-age=31536000; includeSubDomains; preload" always;
+
+    client_max_body_size 20M; # Max size of a file that a user can upload
+
+    # Logs
+    error_log /var/log/nginx/mbin_error.log;
+    access_log /var/log/nginx/mbin_access.log;
+
+    location / {
+        proxy_http_version 1.1;
```
Comment (melroy89) [inline docs/docker_deployment_guide.md:228]: so maybe this is not required, same for the two headers below this `proxy_http_version` line. Out of my head, this is only needed when working with websockets.

Comment (nobodyatroot) [reply docs/docker_deployment_guide.md:228]: yep, you're correct. it's for websocket support, so it's safe to remove.

Comment (melroy89) [reply docs/docker_deployment_guide.md]: Done :)

**Review by nobodyatroot — APPROVED** [review]

**Pull request merged**